modules/basics/src/main/java/com/opengamma/basics/value/HalfUpRounding.java

```diff
@@ -0,0 +1,466 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.basics.value;
+
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableConstructor;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.collect.ArgChecker;
+
+/**
+ * Standard implementation of {@code Rounding} that uses the half-up convention.
+ * <p>
+ * This class implements {@link Rounding} to provide the ability to round a number,
+ * Rounding follows the normal {@link RoundingMode#HALF_UP} convention.
+ * For example, this could be used to round a price to the appropriate market convention.
+ * <p>
+ * Note that rounding a {@code double} is not straightforward as floating point
+ * numbers are based on a binary representation, not a decimal one.
+ * For example, the value 0.1 cannot be exactly represented in a {@code double}.
+ */
+@BeanDefinition(builderScope = "private", cacheHashCode = true)
+public final class HalfUpRounding
+    implements Rounding, ImmutableBean, Serializable {
+
+  /**
+   * The 'None' rounding convention, which applies no rounding.
+   */
+  public static final HalfUpRounding NONE = new HalfUpRounding(256, 1);
+  /**
+   * Cache common roundings.
+   */
+  private static final HalfUpRounding[] CACHE = new HalfUpRounding[16];
+  static {
+    for (int i = 0; i < 16; i++) {
+      CACHE[i] = new HalfUpRounding(i, 0);
+    }
+  }
+
+  /**
+   * The number of decimal places to round to.
+   * <p>
+   * Rounding follows the normal {@link RoundingMode#HALF_UP} convention.
+   * <p>
+   * The value must be from 0 to 256 inclusive. 256 is used to indicate no rounding.
```

Comment:
No rounding is logically and infinite number of decimal places. -1 may be a clearer value, interested in opinions.